### PR TITLE
Add constraints on Page Info URLs

### LINKED ISSUE #8781: [Diaspora] Relative links to Diaspora-local posts fail
Abridged source:
```xml
<status_message>
  ...
  <text>[people of the world](/posts/2576978090fd0138ee4c005056264835)</text>
</status_message>
```
Result:
Screen Shot 2020-06-16 at 14 12 07

Additional notes:
The source shows a labeled link but it got transformed into a link preview. I don't think it should happen with labeled links because content is effectively lost with the link label.

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #8781
Part of #3878

Ok, hear me out.

I wanted to change the Page Info behavior.

During a RC period?

Need some tests to ensure there aren't any glaring regression.

Problem?

The global functions were tightly coupled with remote requests, making unit tests impossible.

Also the Page Info feature was still in `include/`, a folder we decided to deprecate almost 3 years ago.

So I did the only logical thing.

I refactored the whole `include/items.php` into a new class with rewritten workflow to isolate testable bits. The file also contained miscellaneous functions used in other modules/classes, so I moved them out as-is.

We still are in RC, so I didn't completely remove the Page Info global functions because they are used in many parts of the code I can't manually test.

### CODE REVIEW

File: mod/item.php [hunk 1]
```diff
@@ -859,12 +865,112 @@ function item_content(App $a)
...
+				'$confirm_name' => 'confirmed',
+				'$cancel' => DI::l10n()->t('Cancel'),
+			]);
+		}
+		// Now check how the user responded to the confirmation query
+		if (!empty($_REQUEST['canceled'])) {
+			DI::baseUrl()->redirect('display/' . $item['guid']);
 		}
+
+		$is_comment = ($item['gravity'] == GRAVITY_COMMENT) ? true : false;
```
Comment (annando) [inline mod/item.php:931]: This `...  ? true : false;` looks simply ugly.

File: mod/item.php [hunk 2]
```diff
@@ -859,12 +865,112 @@ function item_content(App $a)
...
+		$parentitem = null;
+		if (!empty($item['parent'])){
```
Comment (annando) [inline mod/item.php:933]: Could you add a space before `{`?

File: mod/item.php [hunk 3]
```diff
@@ -859,12 +865,112 @@ function item_content(App $a)
...
+				//NOTREACHED
+			} else {
+				DI::baseUrl()->redirect($return_url);
+				//NOTREACHED
+			}
+		}
+	} else {
+		notice(DI::l10n()->t('Permission denied.'));
+		DI::baseUrl()->redirect('display/' . $item['guid']);
+		//NOTREACHED
```
Comment (annando) [inline mod/item.php:972]: Do we still need this?

File: src/Protocol/Feed.php [hunk 4]
```diff
@@ -37,7 +37,67 @@
...
+	 * @param array $importer
+	 * @param array $contact
+	 * @param       $hub
+	 * @throws ImagickException
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	public static function consume($xml, array $importer, array $contact, &$hub)
+	{
+		if ($contact['network'] === Protocol::OSTATUS) {
+			Logger::log('Consume OStatus messages ', Logger::DEBUG);
```
Comment (annando) [inline src/Protocol/Feed.php:77]: Can you replace all `Logger::log(..., Lugger::DEBUG);` with `Logger::info(`?

**Comment by MrPetovan** [comment]
> Just some standards.

Well, well, well, how the tables have turned! It's alright, I'm in favor of all those changes, but knowing how overboard I can go, I deliberately restrained myself from rewriting everything that I touched. I know how tedious giant diffs can be and I'm trying to keep this in mind.

I had started to remove more functions from `include/items.php` but I decided to delay it after the release.

File: mod/item.php (hunk 3 above)
Comment (MrPetovan) [reply mod/item.php:972]: You tell me.

Comment (annando) [reply mod/item.php:970]: I meant the "notreached" comment.

Comment (MrPetovan) [reply mod/item.php:970]: The `redirect()` still has an `exit` call so it may be valuable.

**Pull request merged**